src/object.rs

```diff
@@ -628,6 +629,20 @@ impl<T: IsA<Object> + SetValue> ObjectExt for T {
         }
     }
 
+    fn has_property_with_type<'a, N: Into<&'a str>>(&self, property_name: N, type_: Type) -> Result<(), BoolError> {
+        let property_name = property_name.into();
+        let ptype = self.get_property_type(property_name);
+
+        match ptype {
+            None => Err(BoolError("Invalid property name")),
+            Some(ptype) => if ptype == type_ {
```

Comment:
Keep it in english so other people can follow as well please (même si je pense que tout le monde aura compris ce que t'as dit / even though I think everyone understood what you wrote). :)